projects/github.com/helmfile/vals/package.yml

```
@@ -0,0 +1,31 @@
+distributable:
+  url: https://github.com/helmfile/vals/archive/refs/tags/v{{version}}.tar.gz
+  strip-components: 1
+
+versions:
+  github: helmfile/vals
+
+provides:
+  - bin/vals
+
+build:
+  dependencies:
+    go.dev: '*'
+  script:
+   - go mod download
+   - mkdir -p "{{ prefix }}"/bin
+   - go build $GOARGS -trimpath -ldflags="$LDFLAGS" -o "{{ prefix }}"/bin/vals ./cmd/vals
+  env:
+    LDFLAGS:
+      - -w
+      - -s
+      - -X main.version=v{{version}}
```

```suggestion
  dependencies:
    go.dev: '*'
    curl.se: '*'
    stedolan.github.io/jq: '*'
    
  script:
    - curl -fsS "https://api.github.com/repos/helmfile/vals/git/tags/v{{version}}" | jq '.object.sha' > .commit_sha
    - go mod download
    - mkdir -p "{{ prefix }}"/bin
    - go build $GOARGS -trimpath -ldflags="$LDFLAGS" -o "{{ prefix }}"/bin/vals ./cmd/vals
  env:
    LDFLAGS:
      - -w
      - -s
      - -X main.version=v{{version}} -X main.commit=$(cat .commit_sha)
```

`vals version` shows the commit sha https://github.com/helmfile/vals/blob/57ae35b137ae8b95b979b262df81433eb349b84a/.goreleaser.yml#L8. What do you think of fetching it from GitHub?

---

that'd be the usual way to do it. i dislike things requiring commit sha, but using the git+https distributable and git-scm.org build dep will do it.

---

your curl > jq is elegant for letting us keep using the tarball source.

---

That was the point of implementing it like that..
What are the available moustache variables available? 
Tags are fetched so maybe the associated Sha can be interpolated too?

---

we don't have a moustache for it. in theory, it would be something returnable, but only for certain types of versions, so it'd be empty a lot of the time. for better or worse, many things are not on github.